components/eamxx/src/share/field/field_alloc_prop.cpp

```diff
@@ -77,6 +77,50 @@ subview (const int idim, const int k, const bool dynamic) const {
   return props;
 }
 
+FieldAllocProp FieldAllocProp::subview(const int idim,
+                                       const int index_beg,
+                                       const int index_end) const {
+  EKAT_REQUIRE_MSG(
+      is_committed(),
+      "Error! Subview requires alloc properties to be committed.\n");
+  EKAT_REQUIRE_MSG(idim < m_layout.rank(),
+                   "Error! Dimension index out of bounds.\n");
+  EKAT_REQUIRE_MSG(index_beg < index_end,
+                   "Error! Slice indices are invalid (non-increasing).\n");
+  EKAT_REQUIRE_MSG(
+      index_beg >= 0 && index_end < m_layout.dim(idim),
+      "Error! Slice index range along the idim dimension is out of bounds.\n");
+
+  // Set new layout basic stuff
+  FieldAllocProp props(m_scalar_type_size);
+  props.m_committed = true;
+  props.m_scalar_type_size = m_scalar_type_size;
+  props.m_layout = m_layout.clone();
+  props.m_layout.reset_dim(idim, index_end - index_beg);
+
+  // we do not currently have the capability for contiguous multi-slice views
+  props.m_contiguous = false;
+
+  props.m_subview_info =
+      SubviewInfo(idim, index_beg, index_end, m_layout.dim(idim));
+
+  // Figure out strides/packs
+  if (idim == (m_layout.rank() - 1)) {
+    // We're slicing the possibly padded dim, so everything else is as in the
+    // layout, and there is no packing
+    // props.m_last_extent = m_layout.dim(idim);
+    props.m_last_extent = index_end - index_beg;
+    props.m_pack_size_max = 1;
+    props.m_alloc_size = m_alloc_size / m_last_extent;
```

Comment:
I don't think m_alloc_size is used in anything other than for allocating the view, so here is probably fine. However, it is odd to see this value. I would have expected `props.m_alloc_size = (m_alloc_size / m_last_extent) * props.m_last_extent`.

That said, I think we should change this to be preserved upon subview. So can you change this to be `props.m_alloc_size = m_alloc_size`? And since it should be the same in both cases of this if, you can put it outside the if statement.